### PR TITLE
Last changes in beta

### PR DESCRIPTION
**Pull request by Zeeerooo**
Changes:
- Bug fixes
- Deleted report pages
- New api method with antenna information
- New api method for terms and condition
- Added antenna signal mean reports
- Added antenna network report generation
- Added application rankings
- New way to add events and their context

### CODE REVIEW

File: app/api/events_view.py [hunk 1]
```diff
@@ -52,46 +74,78 @@ def read_events_from_file():
...
+        db.session.add(event)
 
     try:
         # save and commit events and information to the database
         db.session.commit()
     except Exception as e:
-        app.logger.error("Error adding events to de database " + str(e))
+        app.logger.error("Error adding events to database " + str(e))
         db.session.rollback()
         return "Bad Request", 400
```
Comment (camilog) [inline app/api/events_view.py]: Maybe could be better to return another code, not 400 (to differentiate this type of error with typeError, keyError, UnicodeError, etc)

File: app/api/events_view.py [hunk 2]
```diff
@@ -100,94 +154,85 @@ def set_events_context(jsonvar):
...
     return device, sim, app_version_code
 
 
-def save_traffics_events(events, device, sim, app_version_code):
-    total_events = 0
+def store_traffics_events(events, device_id, sim_serial_number, app_version_code, list_events):
     for event in events:
-        total_events += 1
         event["app_version_code"] = app_version_code
         if event["event_type"] == 2:
```
Comment (camilog) [inline app/api/events_view.py]: Doesn't look better a switch/case here?

File: app/api/events_view.py [hunk 3]
```diff
@@ -272,23 +311,19 @@ def save_gsm_events(events, device, sim, app_version_code):
...
-            add_event_in_db(eventModel, device, sim)
-    return total_events
+        link_observation_with_carrier(eventModel)
+        link_gsm_event_with_antenna(eventModel)
+        vinculate_event_device_sim(eventModel, device_id, sim_serial_number)
+        list_events.append(eventModel)
 
 
-def save_telephony_events(events, device, sim, app_version_code):
+def save_telephony_events(events, device_id, sim_serial_number, app_version_code, list_events):
```
Comment (camilog) [inline app/api/events_view.py]: Is necessary this method?

File: app/models/carrier.py [hunk 4]
```diff
@@ -33,13 +33,17 @@ def add_sim(self, sim):
...
-            carrier = Carrier(mnc=args["mnc"], mcc=args["mcc"], name="Unknown")
-            db.session.add(carrier)
-        return carrier
+        if mnc and mcc:
+            carrier = Carrier.query.filter(Carrier.mnc == mnc, Carrier.mcc == mcc).first()
+            if not carrier:
+                carrier = Carrier(mnc=mnc, mcc=mcc, name="Unknown")
+                db.session.add(carrier)
+                db.session.commit()
+                app.logger.error("New antenna added: mnc:"+str(mnc)+", mcc: "+str(mcc))
```
Comment (camilog) [inline app/models/carrier.py]: error log? Doesn't seem to be an error.

File: app/public/views.py [hunk 5]
```diff
@@ -1,70 +1,30 @@
...
 
-@app.errorhandler(404)
-def page_not_found(error):
-    return render_template('page_not_found.html'), 404
+@app.route("/test_reports")
+def tests_reports():
+    from app.report.reports_generation import monthly_reports_generation
+
+    monthly_reports_generation()
+    return "Reportes generados", 200
```
Comment (camilog) [inline app/public/views.py]: english please

File: app/api/events_view.py (hunk 3 above)
Comment (Zeeerooo) [inline app/api/events_view.py]: Yep, for completeness

**Comment by camilog** [comment]
LGTM

**Comment by sebablasko** [comment]
lgtm

**Pull request merged**